website/src/content/development/forms.mdx

```diff
@@ -0,0 +1,673 @@
+import Text from '@commercetools-uikit/text';
+import Spacings from '@commercetools-uikit/spacings';
+import { docToFormValues, formValuesToDoc } from './conversions';
+import ChannelsForm from './channels-form';
+
+const ChannelsCreate = () => {
+  const languages = useApplicationContext((context) => context.project.languages);
+  const handleSubmit = useCallback(
+    async (formValues) => {
+      const data = formValuesToDoc(formValues);
+      // This would trigger the request, for example a mutation.
+      const result = await createChannel(data);
+      // If successful, show a notification and redirect
+      // to the Channels details page.
+      // If errored, show an error notification.
+    },
+    [createChannel]
+  );
+
+  return (
+    <Spacings.Stack scale="xl">
+      <Text.Headline as="h1">
+        Create a channel
+      </Text.Headline>
+      <ChannelsForm
+        initialValues={docToFormValues(null, languages)}
+        onSubmit={handleSubmit}
+      />
+    </Spacings.Stack>
+  );
+}
+```
+
+In the details page we would need to fetch the data first, then initialize the form with the data.
+
+```js title="channels-details.js"
+import { useCallback } from 'react';
+import { useRouteMatch } from "react-router-dom";
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import Text from '@commercetools-uikit/text';
+import Spacings from '@commercetools-uikit/spacings';
+import LoadingSpinner from '@commercetools-uikit/loading-spinner';
+import useChannels from './use-channels';
+import { docToFormValues, formValuesToDoc } from './conversions';
+import ChannelsForm from './channels-form';
+
+const ChannelsDetails = (props) => {
+  const match = useRouteMatch();
+  const languages = useApplicationContext((context) => context.project.languages);
+  const { data: channel } = useChannels(match.params.id)
+  const handleSubmit = useCallback(
+    async (formValues) => {
+      const data = formValuesToDoc(formValues);
+      // This would trigger the request, for example a mutation.
+      const result = await updateChannel(data);
+      // If successful, show a notification.
+      // If errored, show an error notification.
+    },
+    [updateChannel]
+  );
+
+  if (!channel) {
+    return <LoadingSpinner />;
+  }
+
+  return (
+    <Spacings.Stack scale="xl">
+      <Text.Headline as="h1">
+        Manage Channel
+      </Text.Headline>
+      <ChannelsForm
+        initialValues={docToFormValues(channel, languages)}
+        onSubmit={handleSubmit}
+      />
+    </Spacings.Stack>
+  );
+}
+```
+
+## Using modal pages
+
+Most of the time a create or details page with a form can be implemented using either the [FormModalPage](/api-reference/commercetools-frontend-application-components#formmodalpage) or the [CustomFormModalPage](/api-reference/commercetools-frontend-application-components#customformmodalpage) components.
```

Comment:
I was pretty confused with the "a create or details page" formulation as I thought "create" was a verb. Now I get it that it's a create page or a details page. But maybe this could be formulated a bit better.